### PR TITLE
EventHub: 2018-preview - added new API for NetworkRuleSet 

### PR DESCRIPTION
**Pull request by v-Ajnava**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [X] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (KrisBash) [inline specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: The resource type name should be plural: networkRuleset**s**. I'm not sure what the distinction is between virtualNetworkRules and networkRulsets, but could an improved resource type name help make that distinction?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comment about pluralization of resource type name

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (v-Ajnava) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: resolved

Comment (v-Ajnava) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: resolved

Comment (v-Ajnava) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: as this is the preview version, we will be will be deprecated virtualNetworkRules API and will be going ahead with only networkruleset API. 
for  plural: networkRulesets,  Correct me if I am wrong.  each namespace will be having only one networkRuleset which will consist of ipRule list and virtualNetworkRule list. 
so should we make plural: networkRulesets or singular : networkRuleset ?

**Comment by v-Ajnava** [comment]
@KrisBash can you please see the comment for pluralization. changing the definition name of NetworkRuleset  to loweCamelCase will change class name which Autorest generates.

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (ravbhatnagar) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: yeah, today we do pluralize the resourceType names even though for that type there may be just a single instance possible.

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (nschonni) [inline specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: successfuly -> successfully

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (nschonni) [inline specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: successfuly -> successfully

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (nschonni) [inline specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: successfuly -> successfully

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (nschonni) [inline specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: vaule -> value

Comment (nschonni) [inline specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: vaule -> value

Comment (nschonni) [inline specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: NameSpaceVirtualNetworkRuleSetget -> NameSpaceVirtualNetworkRuleSetGet
Or change it to a sentence

**Review by nschonni — COMMENTED** [review]
minor typos

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (v-Ajnava) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: fixed

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (v-Ajnava) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: fixed

File: specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json
Comment (v-Ajnava) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: fixed

Comment (v-Ajnava) [reply specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json]: fixed, changed the example filenames from NameSpaceVirtualNetworkRuleSetget  to NameSpaceNetworkRuleSetGet

**Comment by ravbhatnagar** [comment]
@v-Ajnava - You have updated the model names to be plural, which was not needed. The resource type name in the URL of the API needs to be pluralized.

**Comment by v-Ajnava** [comment]
@ravbhatnagar  updated resource type name in url with plural and lowerCamelCase and reverted plural model names

**Comment by ravbhatnagar** [comment]
cool, signing off from ARM side.

**Review by praries880 — APPROVED** [review]

**Review by praries880 — APPROVED** [review]

**Comment by praries880** [comment]
@KrisBash You need to approve the changes here. I can not merge the changes in without your approval.

**Review by KrisBash — APPROVED** [review]
Thanks for the changes

**Pull request merged**
